### PR TITLE
Add `micromamba shell activate -n`

### PR DESCRIPTION
**Pull request by jonashaag**
~Remove some dead code and fix bunch of bugs:~

~Fixes https://github.com/mamba-org/mamba/issues/1386~ 
~Fixes https://github.com/mamba-org/mamba/issues/1387~
See #1417 

Fix for https://github.com/mamba-org/mamba/issues/1384 does not actually work, not sure why. `-n` is simply ignored.

### CODE REVIEW

**Comment by wolfv** [comment]
Wow, awesome!
For #1384 there might be an interaction with the shell code added in `micromamba.sh` ... 
Does `micromamba shell activate -n myenv` print anything?

**Comment by wolfv** [comment]
Actually, I just tried, and 

```
micromamba shell activate -n 123123
The following argument was not expected: -n
Run with --help for more information.
```

So maybe that's where the problem lies

**Comment by jonashaag** [comment]
Hm one of us seems to be using a different version of the code :-D

```
./micromamba/micromamba shell activate -n 123
set -gx PATH "/Users/j/micromamba/bin:/opt/homebrew/Caskroom/mambaforge/base/condabin:/Users/j/.volta/bin:/opt/homebrew/bin:/Users/j/dotfiles/sys/bin:/Users/j/.bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/Apple/usr/bin"
set -gx CONDA_PREFIX "/Users/j/micromamba"
set -gx CONDA_SHLVL "2"
set -gx CONDA_DEFAULT_ENV "base"
set -gx CONDA_PROMPT_MODIFIER "(base) "
set -gx CONDA_PREFIX_1 "/opt/homebrew/Caskroom/mambaforge/base"
```

File: micromamba/src/shell.cpp [hunk 1]
```diff
@@ -62,6 +62,11 @@ init_shell_parser(CLI::App* subcom)
             .description("The root prefix to configure (for init and hook), and the prefix "
                          "to activate for activate, either by name or by path"));
     subcom->add_option("prefix,-p,--prefix", prefix.set_cli_config(""), prefix.description());
+
+    auto& name = config.insert(Configurable("shell_name", std::string(""))
```
[the remark this answers is not in the document]
Comment (wolfv) [reply micromamba/src/shell.cpp:66]: sorry, I think I was too quick, maybe not :)

**Comment by wolfv** [comment]
@jonashaag I am wondering if the problem coudl be the quoting here: https://github.com/mamba-org/mamba/blob/32e5056cbf80aa8fae745be102044fd996152036/libmamba/data/micromamba.sh#L42

**Comment by wolfv** [comment]
The remaining issue is `activate -n myenv`, correct?

**Comment by wolfv** [comment]
Should we close this PR in the meantime? Since it partially got superseded by #1417

**Comment by jonashaag** [comment]
> The remaining issue is `activate -n myenv`, correct?

Yes.

Recycled this PR for this issue only.

**Comment by wolfv** [comment]
I think I found a simpler fix :)

Check it out!

**Pull request merged**